core/src/main/java/org/infinispan/marshall/exts/DoubleSummaryStatisticsExternalizer.java

```diff
@@ -28,18 +33,35 @@
    static final Field maxField;
 
    static final boolean canSerialize;
+   static final Constructor<DoubleSummaryStatistics> constructor;
 
    static {
-      countField = SecurityActions.getField(DoubleSummaryStatistics.class, "count");
-      sumField = SecurityActions.getField(DoubleSummaryStatistics.class, "sum");
-      sumCompensationField = SecurityActions.getField(DoubleSummaryStatistics.class, "sumCompensation");
-      simpleSumField = SecurityActions.getField(DoubleSummaryStatistics.class, "simpleSum");
-      minField = SecurityActions.getField(DoubleSummaryStatistics.class, "min");
-      maxField = SecurityActions.getField(DoubleSummaryStatistics.class, "max");
-
-      // We can only properly serialize if all of the fields are non null
-      canSerialize = countField != null && sumField != null && sumCompensationField != null && simpleSumField != null
-              && minField != null && maxField != null;
+      constructor = SecurityActions.getConstructor(
+              DoubleSummaryStatistics.class, long.class, double.class, double.class, double.class);
+
+      if (constructor != null) {
+         // since JDK 10, *SummaryStatistics have parametrized constructors, so reflection can be avoided
+
+         countField = null;
+         sumField = null;
+         sumCompensationField = null;
+         simpleSumField = null;
+         minField = null;
+         maxField = null;
+
+         canSerialize = true;
+      } else {
+         countField = SecurityActions.getField(DoubleSummaryStatistics.class, "count");
+         sumField = SecurityActions.getField(DoubleSummaryStatistics.class, "sum");
+         sumCompensationField = SecurityActions.getField(DoubleSummaryStatistics.class, "sumCompensation");
+         simpleSumField = SecurityActions.getField(DoubleSummaryStatistics.class, "simpleSum");
+         minField = SecurityActions.getField(DoubleSummaryStatistics.class, "min");
+         maxField = SecurityActions.getField(DoubleSummaryStatistics.class, "max");
+
+         // We can only properly serialize if all of the fields are non null
+         canSerialize = countField != null && sumField != null && sumCompensationField != null && simpleSumField != null
+                 && minField != null && maxField != null;
```

Comment:
Do you really need the `sumCompensation` and `simpleSum` fields? If we're ignoring them on JDK10+, we could just ignore them everywhere.

Reply:
Yeah, the method handles are not needed anymore.